src/os/FuseStore.cc

```diff
@@ -594,59 +623,77 @@ static int os_open(const char *path, struct fuse_file_info *fi)
 
   case FN_OBJECT_DATA:
     {
-      pbl = new bufferlist;
-      int r = fs->store->read(cid, oid, 0, 0, *pbl);
+      int r = fs->open_file(
+	path, fi,
+	[&](bufferlist *pbl) {
```

Comment:
maybe we should pass the local variable `bpl` into `open_file()`, because if we go this way, `pbl` will still be `0` at https://github.com/ceph/ceph/pull/10723/commits/604f6b1f494f23d1964e3951ec9169faed5f1eac#diff-fc25a90da304d483f385da5c44750423R686, and hence the returned `fi` will be empty. so the following `os_read()` will run into a null `fi->fh` .